service/routes.py

```diff
@@ -59,159 +59,200 @@ def index():
-
-    app.logger.info("Returning item: %s", inventory.name)
-    return jsonify(inventory.serialize()), status.HTTP_200_OK
-
-
-######################################################################
-#  UPDATE AN INVENTORY ITEM
-######################################################################
-
-@app.route("/inventory/<int:inventory_id>", methods=["PUT"])
-def update_inventory(inventory_id):
-    """
-    Updating an inventory item
-
-    This endpoint will update an item based on the data in the body that is posted
-    """
-
-    app.logger.info("Request to update an inventory item with inventory_id:%s", inventory_id)
-    check_content_type("application/json")
-    item = Inventory.find(inventory_id)
-    if not item:
-        abort(status.HTTP_404_NOT_FOUND, f"Item with inventory_id: {inventory_id} not found")
-    try:
+@api.route('/inventory', strict_slashes=False)
+class InventoryCollection(Resource):
+    """ Handles all interactions with collections of Inventory """
+    ######################################################################
+    # ADD A NEW INVENTORY ITEM
+    ######################################################################
+    @api.doc('create_inventory')
+    @api.response(400, 'The posted data was not valid')
+    @api.expect(create_model)
+    @api.marshal_with(inventory_model, code=201)
+    def post(self):
+        """
+        Creates an inventory item
+        This endpoint will create an item based on the data in the body that is posted
+        """
+        app.logger.info("Request to create an inventory item")
+        check_content_type("application/json")
+        item = Inventory()
         item.deserialize(request.get_json())
-        item.id = inventory_id
-        item.update()
-    except DataValidationError:
-        abort(status.HTTP_400_BAD_REQUEST, "Malformed request")
-
-    return jsonify(item.serialize()), status.HTTP_200_OK
-
+        item.create()
+        location_url = api.url_for(InventoryResource, inventory_id=item.id, _external=True)
+        app.logger.info("Inventory item named [%s] with ID [%s] created.", item.name, item.id)
+        return item.serialize(), status.HTTP_201_CREATED, {'Location': location_url}
+
+    ######################################################################
+    # LIST ALL INVENTORY ITEMS
+    ######################################################################
+    @api.doc('list_inventory_items')
+    @api.expect(inventory_args, validate=True)
+    @api.marshal_list_with(inventory_model)
+    def get(self):
+        """
+        List all inventory items
+
+        This endpoint will list all inventory items in the database
+        """
+        app.logger.info("Request to list all inventory items")
+        items = []
+        condition = request.args.get("condition")
+        restock = request.args.get("restock")
+        quantity = request.args.get("quantity")
+        name = request.args.get("name")
```

Comment:
Get these from inventory_args

```suggestion
        args = inventory_args.parse_args()
```